fedimint-client/src/client/builder.rs

```diff
@@ -1063,30 +1038,43 @@ impl ClientPreview {
     ) -> anyhow::Result<ClientHandle> {
         let pre_root_secret = pre_root_secret.to_inner(self.config.calculate_federation_id());
 
-        let backup = if let Some(backup) = custom_backup {
-            Some(backup)
-        } else {
-            self.inner
-                .download_backup_from_federation(
-                    pre_root_secret.clone(),
-                    &self.config,
-                    self.api_secret.clone(),
-                )
-                .await?
-        };
-
         let client = self
             .inner
             .init(
                 pre_root_secret,
                 self.config,
                 self.api_secret,
                 InitMode::Recover {
-                    snapshot: backup.clone(),
+                    snapshot: custom_backup.clone(),
                 },
             )
             .await?;
 
         Ok(client)
     }
+
+    /// Download most recent valid backup found from the Federation
+    pub async fn download_backup_from_federation(
+        &self,
+        pre_root_secret: RootSecret,
+    ) -> anyhow::Result<Option<ClientBackup>> {
+        let pre_root_secret = pre_root_secret.to_inner(self.config.calculate_federation_id());
+        let api = DynGlobalApi::from_endpoints(
+            // TODO: change join logic to use FederationId v2
+            self.config
+                .global
+                .api_endpoints
+                .iter()
+                .map(|(peer_id, peer_url)| (*peer_id, peer_url.url.clone())),
+            &self.api_secret,
+        )
+        .await?;
+
+        Client::download_backup_from_federation_static(
+            &api,
+            &pre_root_secret,
```

Comment:
```
            &Self::federation_root_secret(&pre_root_secret, config)
```

was lost in this place.

Reply:
You are correct. Verified the logic against 0.7, there's an extra derivation step there as well: https://github.com/fedimint/fedimint/blob/b983d25d4c3cce1751c54e3ad0230fc507e3aeec/fedimint-client/src/client/builder.rs#L407